src/main/java/edu/kit/compiler/semantic/SemanticException.java

```diff
@@ -28,4 +37,12 @@ public SemanticException(String msg, SourceLocation sourceLocation) {
     public JavaEasyCompiler.Result getResult() {
         return JavaEasyCompiler.Result.SemanticError;
     }
+
+    private static String errorListToMsg(Iterable<SemanticError> errors) {
```

Comment:
Why do we need to accumulate the errors? Couldn't we just dump each one into a logger as we go?

Reply:
Do you mean dumping the error when encountering it, or dumping it when constructing the SemanticException? The latter would add a significant side effect to the constructor, which seems rather weird to me

Reply:
I mean dumping the errors as soon as we encounter them. If I see it right, your intention is to have a collection of `Errors` in your class, which you would turn into a single exception at the end of the pass. I would propose some kind of message queue, with a method like `void receive(Error error)`. The implementation of that queue could then for example pass the errors on to a logger, or store them if desired.

Reply:
The reason why I would prefer not to wrap all errors into a single exception is that it would allow us to keep our log output consistent.

Reply:
Okay, I have added a queue as proposed.

Note that the queue has an `hasError`-flag, so we can check later on whether any error occured.